File: game.py
from .gameboard import gameboard, gameboard2
from .parent_node import parent_node
from random import randint
from .computed_boards import computed_boards

class game:
    def __init__(self):
        self.board = gameboard()
        self.computed_boards = computed_boards()
        print("Game has begun")

    def read_move(self):
        while True:
            inp = input("Enter a move\n")
            if len(inp) == 0:
                continue
            m = int(inp)
            if not isinstance(m, int) or m >= 7:
                print("please enter an integer less than 7")
            else:
                self.board.move(m)
                break

    def rand_move(self):
        m = randint(0, len(self.board.valid_moves) - 1)
        self.board.move(self.board.valid_moves[m])

    def computer_move(self):
        node = parent_node(self.board, computed_boards(), max_nodes = self.n_nodes)
        m = node.recommend_move()
        self.board.move(m)


    def start(self, computer_first=False, n_nodes=10_000):
        self.n_nodes = n_nodes
        if computer_first:
            self.board._player_turn = 2
        while not self.board.is_won:
            if self.board.player_turn == 1:
                self.read_move()
            else:
                print("Computer moves")
                self.computer_move()
            print(self.board)
        print("player %d WON!" % (3-self.board._player_turn))

    def start_2player(self):
        while not self.board.is_won:
            print("player %ds turn" % self.board.player_turn)
            self.read_move()
            print(self.board)
        print("player %d WON!" % (3-self.board._player_turn))

File: src/game.jl
include("board.jl")
include("node.jl")


function get_move(board)
    while true
        println("Enter a move")
        inp = readline()
        if length(inp) == 0
            continue
        end

        m = 0
        try
            m = parse(Int, inp)
            if m >= 8 || m <= 0 
                throw(ArgumentError)
            end
        catch ArgumentError
            println("must enter a digit between 0 and 6")
        end

        if m in valid_moves(board)
            return m
        else
            print("invalid move: %m")
        end

    end
end


function rand_move!(board)
    moves = valid_moves(board)
    m = rand(Int) % length(moves) + 1
    move!(board, moves[m])
    return m
end


function computer_move!(board, depth=5)
    boards = ComputedBoards()
    s_n, move = score(board, depth, -Inf, Inf, boards)

    println(s_n)
    println(move)
    move!(board, move)
end


function start_2player()
    board = Gameboard()
    while is_won(board) == 0
        ply = current_player(board) == 1 ? 1 : 2
        println("player $ply turn")
        move = get_move()
        move!(board, move)
        println(board)
    end

    ply =  is_won(board)
    if ply == -1 
        println("player 2 Won")
    elseif  ply == 1 
        println("player 1 won")
    else 
        println("tie?")
    end
end


function start(;computer_first=false, depth=5)
    board = Gameboard()

    if computer_first
        board.turn += 1
    end

    while is_won(board) == 0
        turn = current_player(board)
        if turn == 1
            j = get_move(board)
            move!(board, j)
        else
            println("Computer moves")
            computer_move!(board, depth)
        end
        print(board)
    end
    ply = is_won(board)
    print("player $ply WON!")

end









